### DISCUSSION TITLE
navigator.share on Safari iOS not working with LittleJS

### DISCUSSION THREAD

**Discussion by wyattcheng** [KilledByAPixel/LittleJS#117] [Q&A]
Does anybody have ideas as to why navigator.share doesn't seem to be compatible with LittleJS when running a game on Safari on an iPhone?

Here is some sample code. Here is a bare bones index.html

```
<!DOCTYPE html><head>
    <title>TwistMyWords.game</title>
    </head><body>
    
    <!-- <script src=LittleJS/dist/littlejs.js></script>
     <script src="game.js"></script> -->
    <button id="shareButton" style="position: absolute; bottom: 10%; left: 50%; transform: translateX(-50%); z-index: 1000; font-size: 4vw; padding: 2vh 4vw;">Share</button>
    
    <script>
        document.getElementById('shareButton').addEventListener('click', async () => {
            try {
                await navigator.share({
                    title: 'Simple Share',
                    text: 'Share testing'
                });
                console.log('Content shared successfully');
            } catch (error) {
                console.error('Error sharing content:', error);
            }
        });
    </script>
    
    </body>
    </html>
```
If you load this, you get a simple share button that when clicked, brings up the opportunity to send the sample text to a friend via to OS share sheet.

However, if you add a simple blank LittleJS (let's say you use the game.js from LittleJS/examples/empty) then the share button no longer works.

**Comment by KilledByAPixel** [comment]
I just tried it out by using modifying the button in [the ui example](https://killedbyapixel.github.io/LittleJS/examples/?example=UI+System) to use this code...

```
button1.onClick = ()=>
{
    // Check if Web Share API is available
    if (!navigator.share)
    {
        console.log('Share API not supported');
        return;
    }
    
    navigator.share({
        title: 'LittleJS',
        text: 'Check out this awesome game engine!',
        url: 'https://github.com/KilledByAPixel/LittleJS'
    })
    .then(()=> console.log('Shared successfully'))
    .catch((error)=> 
    {
        // Don't log error if user cancelled
        if (error.name !== 'AbortError')
            console.error('Share failed:', error);
    });
}
```

I can confirm it is working on iPhone in both Safari and Chrome. It is also working in chrome on PC.

It will not work in some browsers that don't support the share API feature, like Firefox is a major one.

https://developer.mozilla.org/en-US/docs/Web/API/Navigator/share

here's what it looks like on pc...

<img width="1920" height="1080" alt="2025-10-28 11_42_05-NVIDIA GeForce Overlay" src="https://github.com/user-attachments/assets/a64b0ca1-9bfa-41ec-bf64-56e76ae916f4" />
